addons/mail/models/mail_activity.py

```diff
@@ -249,11 +265,23 @@ def action_feedback(self, feedback=False):
                 subtype_id=self.env.ref('mail.mt_activities').id,
                 mail_activity_type_id=activity.activity_type_id.id,
             )
-            message |= record.message_ids[0]
 
-            # Removing attachments res_id to prevent unlinking it
-            # We still want them in the message
-            self.env['ir.attachment'].search([('res_model', '=', self._name), ('res_id', 'in', self.ids)]).write({'res_id': False})
+            # Moving the attachments in the message
+            # TODO: Fix void res_id on attachment when you create
+            # an activity with an image directly see (/web_editor/attachment/add
+            activity_message = record.message_ids[0]
+            message_attachments = self.env['ir.attachment'].browse(activity_attachments[activity.id])
```

Comment:
This should also cover empty lists
```suggestion
            self.env['ir.attachment'].browse(activity_attachments[activity.id]).write({
                'res_id': record.message_ids[0].id,
                'res_model': record.message_ids[0]._name,
            })
```